### DISCUSSION TITLE
Zero receiver queue size for multi topic consumers

### DISCUSSION THREAD

**Discussion by AndyKirsch** [apache/pulsar#18944] [Q&A] [answered]
Tried setting the receiver queue size to zero on the java client for a regex subscription and can't. 
https://github.com/apache/pulsar/blame/master/site2/website/versioned_docs/version-2.10.x/cookbooks-message-queue.md#L23 makes it looks like that is a thing I should be able to do. https://github.com/apache/pulsar/blob/master/pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java#L134-L135 on the client side prevents it. I don't understand why this would be different than the single topic version.

**Comment by michaeljmarshall** [comment, marked as answer]
The documentation indicating that it cannot be done for a partitioned topic is going to provide the same justification: https://github.com/apache/pulsar/blame/master/site2/website/versioned_docs/version-2.10.x/cookbooks-message-queue.md#L25.

For a single topic, the zero queue consumer sends a request for a message and waits indefinitely until a message is sent by the broker. In the case of a multi-topic consumer, like the regex consumer, the routing would be overly complex. The consumer would need to try to consume from individual topics, but not at the same time, otherwise the consumer could have more than zero messages in the queue.

The docs could be made clearer though to indicate that any multi-topic consumer (e.g. a partitioned topic consumer, a regex consumer, or a topics list consumer) is not able to use a zero queue consumer.
